src/azure-cli-core/azure/cli/core/context.py

```diff
@@ -0,0 +1,158 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+import os
```

Comment:
Should context.py really be part of CLI core? In theory the entire CLI and the cloud selection works without any "context". If that's the case, it seems to me like it would more appropriately fit in the extensions world, since it is a user convenience.